package/MDAnalysis/coordinates/memory.py

```diff
@@ -360,17 +360,16 @@ def __init__(self, coordinate_array, order='fac',
         self.ts = self._Timestep(self.n_atoms, **kwargs)
         self.ts.dt = dt
         if dimensions is None:
-            self.dimensions_array = dimensions
-        else:
-            if dimensions.shape == (6,):
-                # single box, tile this to trajectory length
-                # allows modifying the box of some frames
-                dimensions = np.tile(dimensions, (self.n_frames, 1))
-            if dimensions.shape != (self.n_frames, 6):
-                raise ValueError("Provided dimensions array has shape {}. "
-                                 "This must be a array of shape (6,) or "
-                                 "(n_frames, 6)".format(dimensions.shape))
-            self.dimensions_array = dimensions
+            dimensions = np.zeros((self.n_frames, 6), dtype=np.float64)
```

Comment:
Did you decide to always have dimensions as an array, even if it contains zeros? (I assume this is more consistent with how `ts.dimensions` works, and it allows setting the box if required – more flexibility!) :+1: